library/options.inc.php

```diff
@@ -122,9 +122,30 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
 	$selectEmptyName = xlt($empty_name);
 	if ($empty_name)
 		$s .= "<option value=''>" . $selectEmptyName . "</option>";
-	$lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY seq, title", array($list_id));
-	$got_selected = FALSE;
-	
+
+    // List order depends on language translation options.
+    $lang_id = empty($_SESSION['language_choice']) ? '1' : $_SESSION['language_choice'];
+
+    if ($GLOBALS['gb_how_sort_list'] == '0') {
+      //sort by seq
+      $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ?  AND activity=1 ORDER BY seq, title", array($list_id));
+    } elseif($GLOBALS['gb_how_sort_list'] == '1') {
+      // sort by title
+      if (($lang_id == '1' && !empty($GLOBALS['skip_english_translation'])) || !$GLOBALS['translate_lists']) {
+        $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ?  AND activity=1 ORDER BY title, seq", array($list_id));
+      } else {
+        $lres = sqlStatement("SELECT lo.option_id, " .
+            "IF(LENGTH(ld.definition),ld.definition,lo.title) AS title " .
+            "FROM list_options AS lo " .
+            "LEFT JOIN lang_constants AS lc ON lc.constant_name = lo.title " .
+            "LEFT JOIN lang_definitions AS ld ON ld.cons_id = lc.cons_id AND " .
+            "ld.lang_id = ? " .
+            "WHERE lo.list_id = ?  AND lo.activity=1 " .
+            "ORDER BY IF(LENGTH(ld.definition),ld.definition,lo.title), lo.seq", array($lang_id, $list_id));
+      }
+    }
+    $got_selected = FALSE;
+
```

Comment:
Let me look into this. I think you will only need to above block to get what you want and would then skip the translation below since already done.